scripts/log_tool.py

```diff
@@ -0,0 +1,75 @@
+#!/usr/bin/env python3
+import argparse
+import requests
+import sys
+
+
+LEVELS = ('trace', 'debug', 'info', 'warning', 'error', 'critical')
+
+
+def _level(text):
+    if text is None:
+        return None
+
+    candidates = []
+    for candidate in LEVELS:
+        if text.lower() in candidate:
+            candidates.append(candidate)
+
+    if len(candidates) != 1:
+        print('Unable to determine level')
+        sys.exit(1)
+
+    return candidates[0]
+
+
+def parse_commandline():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('logger', nargs='?', help='The target logger')
+    parser.add_argument('level', nargs='?', type=_level,
```

Comment:
I hadn't seen this functionality before - type is a closure?

Reply:
Yep I quite like this trick too. It basically works because in python calling `int(value)` as a constructor is the same syntax as `functor(value)`